machine/v1beta1/types_machine.go

```diff
@@ -360,6 +388,23 @@ type MachineStatus struct {
 
 	// Conditions defines the current state of the Machine
 	Conditions Conditions `json:"conditions,omitempty"`
+
+	// authoritativeAPI is the API that is authoritative for this resource.
+	// Valid values are MachineAPI, ClusterAPI and Migrating.
+	// This value is updated by the migration controller to reflect the authoritative API.
+	// Machine API and Cluster API controllers use this value to determine whether or not to reconcile the resource.
+	// When set to Migrating, the migration controller is currently performing the handover of authority from one API to the other.
+	// +kubebuilder:validation:Enum=MachineAPI;ClusterAPI;Migrating
+	// +openshift:enable:FeatureGate=MachineAPIMigration
+	// +optional
+	AuthoritativeAPI MachineAuthority `json:"authoritativeAPI,omitempty"`
+
+	// synchronizedGeneration is the generation of the authoritative resource that the non-authoritative resource is synchronised with.
+	// This field is set when the authoritative resource is updated and the sync controller has updated the non-authoritative resource to match.
+	// +kubebuilder:validation:Minimum=0
```

Comment:
Generation on the MAPI vs CAPI resources may be different, so during a transition it may go down. This always represents the state of the sync, no matter which side is authoritative

Reply:
> Generation on the MAPI vs CAPI resources may be different, so during a transition it may go down. This always represents the state of the sync, no matter which side is authoritative

So it goes something like
1. authoritative is MAPI, value only goes up
2. authoritative is migration, value is reset (may go down)
3. authoritative is CAPI, value only goes up.

is that accurate?  Can we validate it? (this is a decently likely bug).

Reply:
Yes I think that's approximately accurate.

Part of the workflow requires, while the value is `Migrating`, that we wait for the sync gen field to be up to date. So the workflow could look like
* spec.Auth: MAPI, status.Auth MAPI: syncedGen monotonic
* spec.Auth: CAPI, status.Auth -> Migrating: syncedGen updated final time by sync controller
* Once syncedGen is up to date, status.Auth -> CAPI and syncedGen -> 0 in one call
* spec.Auth: CAPI, status.Auth CAPI: syncGen monotonic again.

So I'm thinking the validation should be
```
self.synchronizedGeneration >= oldSelf.synchonizedGeneration || (oldSelf.authoritativeAPI == 'Migrating' && self.authoritatitiveAPI != 'Migrating')
```
So, either it's monotonic or we just moved from Migrating to another state